AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Storage/SQLite/Model+SQLite.swift

```diff
@@ -23,24 +23,25 @@ extension Persistable {
     /// - Returns: the value as `Binding`
     internal func asBinding() -> Binding {
         let value = self
-
-        if let value = value as? Bool {
-            return Int(value.datatypeValue)
-        }
-
-        // if value conforms to binding, resolve it
-        if let value = value as? Binding {
-            return value
-        }
-
-        if let value = value as? Date {
-            return value.iso8601String
+        let valueType = type(of: value)
+        let valueConverter = SQLiteModelValueConverter()
+        do {
+            let binding = try valueConverter.convertToTarget(from: value,
+                                                             fieldType: .from(type: valueType))
+            guard let validBinding = binding else {
+                preconditionFailure("""
+                Converting \(String(describing: value)) of type \(String(describing: valueType))
+                to a SQLite Binding returned a nil value. This is likely a bug in the
+                SQLiteModelValueConverter logic.
```

Comment:
I think we have a canned "search for issues or file one on our GitHub page" string that would be good to append to this

Reply:
we have this one https://github.com/aws-amplify/amplify-ios/blob/461c3cfd8923fbbb6eb10b60495b0f4594879cf4/Amplify/Core/Support/AmplifyErrorMessages.swift